File: project3/modules/school.cpp
#include <iostream>

#include <floor.h>
#include <stairs.h>
#include <yard.h>
#include <corridor.h>
#include <class.h>
#include <school.h>

#define FLOORS 3
#define CLASSES 6

School::School(const unsigned sizeClass){
    std::cout<<"A new school is being created\n";
    
    yard= new Yard();
    stairs= new Stairs();

    floors =new Floor*[3];
    for(int i=0;i<FLOORS;i++)
        floors[i]= new Floor(i,sizeClass);

}

School::~School(){
    for(int i=0;i<FLOORS;i++)
        delete floors[i];
    delete[] floors;
    delete stairs;
    delete yard;
    std::cout <<"A school to be destroyed\n";
}

Yard& School::getYard() const{
    return *yard;
}
Stairs& School::getStairs()const{
    return *stairs;
}
const Floor& School::getFloor(int floor) const{
    return *floors[floor];
}  
Corridor& School::getCorridor(int floor)const{
    return floors[floor]->getCorridor();
}
TheClass& School::getClass(int floor, int number) const{
    return floors[floor]->getClass(number);
}

void School::print() const{
    std::cout<<"Schoollife consists of: \n";
    yard->print();
    stairs->print();
    for(int i=0;i<FLOORS;i++)
        floors[i]->print();
}

void School::enter(Student* student) {
    yard->enter(student);
}

void School::operate(int hours){
    for(int i=0;i<FLOORS;i++)
        for(int j=0;j<CLASSES;j++)
            floors[i]->getClass(j).operate(hours);
}

File: project4/src/schoolComplex/School.java
package schoolComplex;

import java.util.Vector;
import People.Student;
public class School {

    private Vector<Floor> floors;
    private Stair stair;
    private Yard yard; 

    public School(int maxStudentClass){
        floors = new Vector<Floor>(3);
        for (int i=0;i<3;i++)
            floors.add(new Floor(i,maxStudentClass));
        stair=new Stair();
        yard= new Yard();
    }
    
    public Yard getYard() {
        return yard;
    }
    public Stair getStair(){
        return stair;
    }
    public Floor getFloor(int index){
        return floors.get(index);
    }
    public Corridor getCorridor(int index){
        return floors.get(index).getCorridor();
    }
    public TheClass getClass(int index, int number){
        return floors.get(index).getClass(number);
    } 

    public void print() {
        System.out.print("SchoolComplex:");
        yard.print();
        stair.print();
        for(int i=0;i<3;i++){
            floors.get(i).print();
        } 
    }

    public void enter(Student student,Boolean way){
        yard.enter(student,way);
    }
    
    public void empty(){
        for (int i=0;i<3;i++){
            for(int j=0;j<6;j++){
                Student student=floors.get(i).getClass(j).exit();
                while(student!=null){
                    floors.get(i).getCorridor().enter(student,false);
                    floors.get(i).getCorridor().exit(student,false);
                    stair.enter(student,false);
                    stair.exit(student,false);
                    yard.enter(student,false);
                    yard.exit(student,false);                    
                    student=floors.get(i).getClass(j).exit();
		}
                floors.get(i).getClass(j).exitTeacher();
            }
        }
    }

    public void operate(int hours){
        for (int i=0;i<3;i++)
            floors.get(i).operate(hours);
    }
}
